### PR TITLE
feat: content source section

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Add content source section.
It re-uses the search capability.
The source list is new
User list is re-used, but needs some final styling probably.

@capJavert Also solves: https://dailydotdev.atlassian.net/jira/software/projects/AS/boards/30?selectedIssue=AS-851
(+ tracking additions from here: https://dailydotdev.slack.com/archives/C06LKPZAZ1T/p1734444951948489)
Just need to validate in the end once we add to my feed how it does it there.

(Also test needed for invalidation still)
Screenshot 2024-12-17 at 14 38 10
Screenshot 2024-12-17 at 14 38 05

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-846 #done 

### Preview domain
https://as-846-content-sources.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/feeds/FeedSettings/FeedSettingsEditContext.tsx [hunk 1]
```diff
@@ -1,5 +1,8 @@
-import { createContext } from 'react';
+import { createContext, useContext } from 'react';
 import { FeedSettingsEditContextValue } from './types';
 
 export const FeedSettingsEditContext =
   createContext<FeedSettingsEditContextValue>(undefined);
+
+export const useFeedSettingsEditContext = (): FeedSettingsEditContextValue =>
```
Comment (rebelchris) [inline packages/shared/src/components/feeds/FeedSettings/FeedSettingsEditContext.tsx:7]: I just prefer these :)

File: packages/shared/src/components/feeds/FeedSettings/components/FollowingSourceList.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    id: user.id,
+    entity: ContentPreferenceType.Source,
+    feedId: feed.id,
+  });
+
+  const { data, isFetchingNextPage, fetchNextPage } = queryResult;
+  const sources = useMemo(() => {
+    return data?.pages.reduce((acc, p) => {
+      p?.edges.forEach(({ node }) => {
+        if (type && node.source.type === type) {
```
Comment (rebelchris) [inline packages/shared/src/components/feeds/FeedSettings/components/FollowingSourceList.tsx:29]: One list to render 2 types :)

File: packages/shared/src/components/feeds/FeedSettings/components/FollowingSourceList.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }, [data, type]);
+
+  if (queryResult.isPending) {
+    return null;
+  }
+
+  return (
+    <SourceList
+      sources={sources}
+      emptyPlaceholder={<p>Can't find any sources</p>}
```
Comment (rebelchris) [inline packages/shared/src/components/feeds/FeedSettings/components/FollowingSourceList.tsx:45]: Waiting product info

**Review by capJavert — COMMENTED** [review]
Awesome work!

Don't see anything wrong code wise, functionality wise, I see optimistic update when following during search or listing:
- remove followed source (does not remove from list)
- follow (add) from search, goes to source page (should stay on modal?)

**Review by capJavert — APPROVED** [review]

**Comment by capJavert** [comment]
Still small issue with invalidation but merging to unblock, we will get it tidy down the line.

**Pull request merged**